bluebottle/suggestions/serializers.py

```diff
@@ -10,13 +10,16 @@ def from_native(self, value):
         except IndexError:
             return value
 
-class PublicSuggestionSerializer(serializers.ModelSerializer):
-    deadline = DateField()
 
+import uuid
+
+def generate_token():
+    return str(uuid.uuid4())
+
+class SuggestionSerializer(serializers.ModelSerializer):
     class Meta:
         model = Suggestion
-        exclude = ('token', )
 
-
-class SuggestionSerializer(PublicSuggestionSerializer):
-    project = serializers.SlugRelatedField(slug_field='slug')
+    deadline = DateField()
+    token = serializers.CharField(required=False, default=generate_token)
```

Comment:
email validation. Once a suggestion has been created the frontend app (booking.com) sends out an email to the provided email address with a verification link containing this code. This token is then passed back to the bluebottle suggestion token validation api endpoint which changes the suggestion's state to 'draft'